token/cli/src/main.rs

```diff
@@ -3071,6 +3181,208 @@ async fn command_enable_disable_confidential_transfers(
+    let token = token_client_from_config(config, &state_with_extension.base.mint, None)?;
+
+    // the amount the user wants to deposit or withdraw, as an f64
+    let maybe_amount =
+        ui_amount.map(|ui_amount| spl_token::ui_amount_to_amount(ui_amount, mint_info.decimals));
+
+    // the amount we will deposit or withdraw, as a u64
+    let amount = if !config.sign_only && instruction_type == ConfidentialInstructionType::Deposit {
+        let current_balance = token
+            .get_account_info(&token_account_address)
+            .await?
+            .base
+            .amount;
+        let deposit_amount = maybe_amount.unwrap_or(current_balance);
+
+        println_display(
+            config,
+            format!(
+                "Depositing {} confidential tokens",
+                spl_token::amount_to_ui_amount(deposit_amount, mint_info.decimals),
+            ),
+        );
+
+        if deposit_amount > current_balance {
+            return Err(format!(
+                "Error: Insufficient funds, current balance is {}",
+                spl_token_2022::amount_to_ui_amount_string_trimmed(
+                    current_balance,
+                    mint_info.decimals
+                )
+            )
+            .into());
+        }
+
+        deposit_amount
+    } else if !config.sign_only && instruction_type == ConfidentialInstructionType::Withdraw {
+        // // TODO: expose account balance decryption in token
+        // let aes_key = aes_key.expect("AES key must be provided");
+        // let current_balance = token
+        //     .confidential_transfer_get_available_balance_with_key(&token_account_address, aes_key)
+        //     .await?;
+        let withdraw_amount =
+            maybe_amount.expect("ALL keyword is not currently supported for withdraw");
+
+        println_display(
+            config,
+            format!(
+                "Withdrawing {} confidential tokens",
+                spl_token::amount_to_ui_amount(withdraw_amount, mint_info.decimals)
+            ),
+        );
+
+        withdraw_amount
+    } else {
+        maybe_amount.unwrap()
+    };
+
+    let res = if instruction_type == ConfidentialInstructionType::Deposit {
+        token
+            .confidential_transfer_deposit(
+                &token_account_address,
+                &owner,
+                amount,
+                decimals,
+                &bulk_signers,
+            )
+            .await?
+    } else if instruction_type == ConfidentialInstructionType::Withdraw {
+        let elgamal_keypair = elgamal_keypair.expect("ElGamal keypair must be provided");
+        let aes_key = aes_key.expect("AES key must be provided");
+
+        token
+            .confidential_transfer_withdraw(
+                &token_account_address,
+                &owner,
+                None,
+                amount,
+                decimals,
+                None,
```

Comment:
Can you pass the `WithdrawAccountInfo` here based on `state_with_extension` that was fetched earlier to avoid a second fetch?

Reply:
Yes, that is a great idea.